### DISCUSSION TITLE
Select count where a field is contained within a given string

### DISCUSSION THREAD

**Discussion by Zenoo** [prisma/prisma#22334] [Q&A] [answered]
### Question

Hello,

Is it possible to get the rows count where a string field is contained within a given string value?

Basically I want to inverse of this query:

```ts
const banned = await prisma.bannedWord.count({
  where: {
    word: { contains: req.body.name.toLowerCase() },
  },
});
```

Something like
```ts
const banned = await prisma.bannedWord.count({
  where: {
    word: { isContainedIn: req.body.name.toLowerCase() },
  },
});
```

The equivalent in SQL is:
```sql
SELECT COUNT(*) FROM "BannedWord" WHERE 'someTextIncludingaBadWord' LIKE CONCAT('%', word, '%')
```

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @Zenoo 👋 

Thank you for raising this question.

We do not currently support a direct way to filter records where a string field is contained within another given string value using the `isContainedIn` operator. The standard `contains` filter in Prisma checks if the field contains the given string, not the other way around.

However, there is a GitHub [issue](https://github.com/prisma/prisma/issues/7589) for supporting this particular functionality. To implement this, you will need to use a raw SQL query to achieve this functionality. Here is how you could write a raw SQL query using Prisma's `$queryRaw` method:
```
const bannedCount = await prisma.$queryRaw`SELECT COUNT(*) FROM "BannedWord" WHERE ${req.body.name.toLowerCase()} LIKE CONCAT('%', word, '%')`;
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
